R-package/src/cmake/modules/FindLibR.cmake

```diff
@@ -0,0 +1,229 @@
+    endif()
+  endif()
+
+  if(UNIX)
+
+    if(NOT LIBR_EXECUTABLE)
+      message(FATAL_ERROR "Unable to locate R executable.\
+        \nEither add its location to PATH or provide it through the LIBR_EXECUTABLE cmake variable")
+    endif()
+
+    # ask R for the home path
+    execute_process(
+      COMMAND ${LIBR_EXECUTABLE} "--slave" "--vanilla" "-e" "cat(R.home())"
+      OUTPUT_VARIABLE LIBR_HOME
+    )
+
+    # ask R for the include dir
+    execute_process(
+      COMMAND ${LIBR_EXECUTABLE} "--slave" "--no-save" "-e" "cat(R.home('include'))"
+      OUTPUT_VARIABLE LIBR_INCLUDE_DIRS
+    )
+
+    # ask R for the lib dir
+    execute_process(
+      COMMAND ${LIBR_EXECUTABLE} "--slave" "--no-save" "-e" "cat(R.home('lib'))"
+      OUTPUT_VARIABLE LIBR_LIB_DIR
+    )
+
+  # Windows
+  else()
+
+    # ask R for R_HOME
+    if(LIBR_EXECUTABLE)
+      execute_process(
+        COMMAND ${LIBR_EXECUTABLE} "--slave" "--no-save" "-e" "cat(normalizePath(R.home(),winslash='/'))"
+        OUTPUT_VARIABLE LIBR_HOME
+      )
+    endif()
+
+    # if R executable not available, query R_HOME path from registry
+    if(NOT LIBR_HOME)
+
+      # Try to find R's location in the registry
+      # ref: https://cran.r-project.org/bin/windows/base/rw-FAQ.html#Does-R-use-the-Registry_003f
+      get_filename_component(
+        LIBR_HOME
+        "[HKEY_LOCAL_MACHINE\\SOFTWARE\\R-core\\R\\${CMAKE_R_VERSION};InstallPath]"
+        ABSOLUTE
+      )
+
+      if(NOT LIBR_HOME)
+        get_filename_component(
+          LIBR_HOME
+          "[HKEY_CURRENT_USER\\SOFTWARE\\R-core\\R\\${CMAKE_R_VERSION};InstallPath]"
+          ABSOLUTE
+        )
+      endif()
+
+      if(NOT LIBR_HOME)
+        message(FATAL_ERROR "\nUnable to locate R executable.\
+          \nEither add its location to PATH or provide it through the LIBR_EXECUTABLE cmake variable")
+      endif()
+
+    endif()
+
+    # set exe location based on R_ARCH
+    if(NOT LIBR_EXECUTABLE)
+      set(LIBR_EXECUTABLE "${LIBR_HOME}/bin/${R_ARCH}/R.exe")
+    endif()
+
+    # set other R paths based on home path
+    set(LIBR_INCLUDE_DIRS "${LIBR_HOME}/include")
+    set(LIBR_LIB_DIR "${LIBR_HOME}/bin/${R_ARCH}")
+
+    message(STATUS "LIBR_HOME [${LIBR_HOME}]")
+    message(STATUS "LIBR_EXECUTABLE [${LIBR_EXECUTABLE}]")
+    message(STATUS "LIBR_INCLUDE_DIRS [${LIBR_INCLUDE_DIRS}]")
+    message(STATUS "LIBR_LIB_DIR [${LIBR_LIB_DIR}]")
+    message(STATUS "LIBR_CORE_LIBRARY [${LIBR_CORE_LIBRARY}]")
+
+  endif()
+
+endif()
+
+if(WIN32 AND MSVC)
+
+  # create a local R.lib import library for R.dll if it doesn't exist
+  if(NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/R.lib")
+    create_rlib_for_msvc()
+  endif()
+
+endif()
+
+# look for the core R library
+find_library(LIBR_CORE_LIBRARY NAMES R
+  HINTS "${CMAKE_CURRENT_BINARY_DIR}" "${LIBR_LIB_DIR}" "${LIBR_HOME}/bin" "${LIBR_LIBRARIES}")
+
+if(LIBR_CORE_LIBRARY-NOTFOUND)
+  message(STATUS "Could not find R core shared library.")
```

Comment:
Is this library required?

Reply:
hmmmm let me try without it, it might not be

Reply:
wait yes, I think it is needed by the linker.

https://github.com/jameslamb/LightGBM/blob/feat/r-header/CMakeLists.txt#L311

Reply:
Maybe move logging part to CMakeLists as per https://github.com/microsoft/LightGBM/pull/2901#discussion_r393696132?

As this library is required, building process will fail without it anyway. Seems that this message doesn't make a lot of sense.

Reply:
ah ok true! Yes I can do that.

Reply:
Actually I just decided to get rid of this message, I think you're right that it's not providing any value

Reply:
removed in https://github.com/microsoft/LightGBM/commit/9adc7a94fafeacf29e0bd88ffa320299406bbd65